azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -597,22 +864,16 @@ public String submitExecutableFlow(ExecutableFlow exflow, String userId)
       // We create an active flow reference in the datastore. If the upload
       // fails, we remove the reference.
       ExecutionReference reference =
-          new ExecutionReference(exflow.getExecutionId(), executorHost,
-              executorPort);
+        new ExecutionReference(exflow.getExecutionId());
+      // Added to db queue
       executorLoader.addActiveExecutableReference(reference);
-      try {
-        callExecutorServer(reference, ConnectorParams.EXECUTE_ACTION);
-        runningFlows.put(exflow.getExecutionId(),
-            new Pair<ExecutionReference, ExecutableFlow>(reference, exflow));
-
-        message +=
-            "Execution submitted successfully with exec id "
-                + exflow.getExecutionId();
-      } catch (ExecutorManagerException e) {
-        executorLoader.removeActiveExecutableReference(reference.getExecId());
-        throw e;
-      }
-
+      Pair<ExecutionReference, ExecutableFlow> pair =
+        new Pair<ExecutionReference, ExecutableFlow>(reference, exflow);
+      queuedFlowMap.put(exflow.getExecutionId(), pair);
```

Comment:
want to create a queue and dequeue method for accessing the Map and the List? this way we can make sure people will not forget to do one or the other operation